src/test/java/net/sf/jabref/model/database/BibDatabaseTest.java

```diff
@@ -317,4 +320,52 @@ public void resolveForStringsOddHashMarkAtTheEnd() {
         database.addString(string);
         assertEquals(database.resolveForStrings("AAA#AAA#AAA#"), "AAAaaaAAA#");
     }
+
+    @Test
+    public void getUsedStringsSingleStringWithString() {
+        BibEntry entry = new BibEntry(IdGenerator.next());
+        entry.setField("author", "#AAA#");
+        BibtexString string = new BibtexString(IdGenerator.next(), "AAA", "Some other #BBB#");
+        BibtexString string2 = new BibtexString(IdGenerator.next(), "BBB", "Some more text");
+        BibtexString string3 = new BibtexString(IdGenerator.next(), "CCC", "Even more text");
+        database.addString(string);
+        database.addString(string2);
+        database.addString(string3);
+        database.insertEntry(entry);
+        List<BibtexString> usedStrings = (List<BibtexString>) database.getUsedStrings(Arrays.asList(entry));
+        assertEquals(2, usedStrings.size());
+        assertTrue((string.getName() == usedStrings.get(0).getName())
```

Comment:
That might indeed be a better solution. The problem at the moment is really that I do not know which order they are added in (or rather, I want the test to be independent of that).